### DISCUSSION TITLE
Is it possible to filter daughters in decay graphs by activity?

### DISCUSSION THREAD

**Discussion by php1ic** [radioactivedecay/radioactivedecay#103] [General]
The below snippet gives the shown output
```
import radioactivedecay as rd
inv = rd.Inventory({ 'Pu-239': 10.0 }, 'Bq')
inv.plot(10, 'm', yunits='Bq', yscale='log')
```

![image](https://github.com/radioactivedecay/radioactivedecay/assets/3843619/45446280-a246-4d60-abdf-326464b13b50)

I can limit the y-axis with the `ymin` option in `plot()` to not see the data below 1e-19Bq,  but that doesn't remove the daughter isotopes from the key.

Is there a way to specify an activity and have the module not display the data for any isotope that never has a value higher than that over the time period specified?

**Comment by alexmalins** [comment]
Hey @php1ic 👋 Perhaps the easiest way to achieve what you'd like is to use the `display` parameter to limit to just the nuclides you want to show. Those can be found roughly by doing one decay and filtering out results with activity below 1E-18 Bq:

```
inv = rd.Inventory({ 'Pu-239': 10.0 }, 'Bq')
decayed_inv = inv.decay(10, 'm')
keep_nuclides = [nuclide for nuclide, activity in decayed_inv.activities().items() if activity > 1E-18]
inv.plot(10, 'm', yunits='Bq', yscale='log', display=keep_nuclides, ymin=1E-19, ymax=100)
```
![keep_nuclides](https://github.com/radioactivedecay/radioactivedecay/assets/22991362/a824206d-b7d7-43dd-887d-25fe39323421)

In this case the activities below 1E-18 look only to be numerical noise anyway, so it makes sense to filter those lines out.

FYI if you'd used the high precision inventory that cleans up all the noise and produces realistic curves:
```
inv_hp = rd.InventoryHP({ 'Pu-239': 10.0 }, 'Bq')]
inv_hp.plot(10, 'm', yunits='Bq', yscale='log', ymax=100)
```
![hp_no_noise](https://github.com/radioactivedecay/radioactivedecay/assets/22991362/22c5e0fc-068c-4d93-8eef-ef4d533de110)

However you may appreciate not having the nuclides with trace quantities plotted, and you can use the solution using `display` outlined above to remove them.

Hope this helps!

**Comment by php1ic** [comment]
That's great, thanks @alexmalins.

There's something at the back of my mind that is unsure about using the final activity as the value to check against, but what you posted does exactly what I need for this isotope. If I come up with a better idea I'll make a PR.

Thanks again!

**Reply by alexmalins** [reply]
Hey sounds great!

> There's something at the back of my mind that is unsure about using the final activity as the value to check against
You're right this won't always do the job.

What you could consider is adding a boolean parameter to the plot method to remove nuclides from the plot if their activities are always below `ymin`, i.e. the scenario described where they appear on the legend but the lines never show on the graph.

This should be simple enough to implement as the [y-values](https://github.com/radioactivedecay/radioactivedecay/blob/main/radioactivedecay/inventory.py#L939) to make a graph are passed towards matplotlib as a (npoints x num_nuclides) matrix. If one of the columns of the matrix (i.e. the activities for one nuclide over time) are always less then ymin, then drop that nuclide from the collection that is plotted.

PR to do this would be very welcome 🙌

**Reply by alexmalins** [reply]
> What you could consider is adding a boolean parameter to the plot method to remove nuclides from the plot if their activities are always below ymin, i.e. the scenario described where they appear on the legend but the lines never show on the graph.

In fact maybe it should just be the default to not try and plot nuclides that won't appear on the graph - i.e. not even adding an additional parameter for this 🤔
